src/libs/workspace-utils.ts

```diff
@@ -29,6 +29,11 @@ interface BaseWorkspaceInfo {
   createdDate: string;
   createdBy: string;
   lastModified: string;
+  bucketName?: string;
+  attributes?: Record<string, unknown>;
+  isLocked?: boolean;
+  canShare?: boolean;
+  state?: WorkpaceState;
```

Comment:
Since existing fields are not marked optional, I would still prefer to follow that convention and not mark these fields optional. I'd rather components props specify the slice of the workspace that they require vs making every field optional and having to deal with that in every component.

Reply:
I'd like to clean this up in a more systematic way, separately.

There's a bunch of different places where we fetch workspaces with different fields, and I'd like to have each one to just use a type that specifies what fields it's using.

But trying to untangle that right now is really messy, since it requires redefining a number of types, and not all of those places have been converted to typescript.

Reply:
> There's a bunch of different places where we fetch workspaces with different fields, and I'd like to have each one to just use a type that specifies what fields it's using.

That's the end goal I have in mind as well. I'm just really not keen on this intermediate state where like half the fields are marked required and half aren't.

Would it be possible to define the type without optional fields, then use Partial or DeepPartial where necessary as workarounds? I think that would give us a more solid foundation to build those specific types for subsets of workspace fields.

Reply:
I like the _idea_ of that, but it breaks the assumptions made in a lot of unrelated areas of the code (and requires marking parameters in all of those areas as Partial).

And if you think about it, they are functioning as optional values - including from the API side, where we're requesting some values in addition to the common ones.

I think I've removed all of the ones values that were straight up incorrect for them to be there. 

It seems like it will be just as easy in the future to delete the fields, then everywhere the fields are used, replace the type with the new one.